File: app/src/main/java/ua/leonidius/rtlnotepad/dialogs/ConfirmEncodingChangeDialog.java
package ua.leonidius.rtlnotepad.dialogs;
import android.app.*;
import android.content.*;
import android.os.*;
import ua.leonidius.rtlnotepad.*;

public class ConfirmEncodingChangeDialog extends DialogFragment implements AlertDialog.OnClickListener
{
	private Callback callback;
	private Activity activity;

	public ConfirmEncodingChangeDialog(Activity activity) {
		super();
		this.activity = activity;
	}

	@Override
	public Dialog onCreateDialog(Bundle savedInstanceState)
	{
		AlertDialog.Builder adb = new AlertDialog.Builder(activity);
		adb.setMessage(R.string.encoding_change_warning);
		adb.setNegativeButton(R.string.no, this);
		adb.setPositiveButton(R.string.yes, this);
		adb.setNeutralButton(R.string.cancel, this);
		return adb.create();
	}

	@Override
	public void onClick(DialogInterface p1, int id)
	{
		switch (id) {
			case Dialog.BUTTON_NEGATIVE:
				callback.callback(Callback.DONT_CHANGE);
				break;
			case Dialog.BUTTON_POSITIVE:
				callback.callback(Callback.CHANGE);
				break;
		}
	}

	public void setCallback(Callback callback) {
		this.callback = callback;
	}

	public interface Callback {
		public static final byte CHANGE = 0x00, DONT_CHANGE = 0x01;

		public void callback(byte response);
	}

}

File: app/src/main/java/ua/leonidius/rtlnotepad/dialogs/ConfirmEncodingChangeDialog.kt
package ua.leonidius.rtlnotepad.dialogs

import android.app.AlertDialog
import android.app.Dialog
import android.content.DialogInterface
import android.os.Bundle
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import ua.leonidius.rtlnotepad.R

class ConfirmEncodingChangeDialog : BaseDialog(), DialogInterface.OnClickListener {

    private lateinit var viewModel: Model

    companion object {
        fun create(callback : (Boolean) -> Unit) : ConfirmEncodingChangeDialog {
            val dialog = ConfirmEncodingChangeDialog()
            dialog.initializerFunction = {
                dialog.getViewModel().callback = callback
            }
            return dialog
        }
    }

    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
        val adb = AlertDialog.Builder(activity)
        adb.setMessage(R.string.encoding_change_warning)
        adb.setNegativeButton(R.string.no, this)
        adb.setPositiveButton(R.string.yes, this)
        adb.setNeutralButton(R.string.cancel, this)
        return adb.create()
    }

    override fun onClick(p1: DialogInterface, id: Int) {
        when (id) {
            Dialog.BUTTON_NEGATIVE -> getViewModel().callback(false)
            Dialog.BUTTON_POSITIVE -> getViewModel().callback(true)
        }
    }

    fun getViewModel() : Model {
        if (!this::viewModel.isInitialized) {
            viewModel = ViewModelProvider(this).get(Model::class.java)
        }
        return viewModel
    }

    class Model : ViewModel() {
        lateinit var callback : (Boolean) -> Unit
    }

}